dkp/dkp.py

```diff
@@ -267,22 +304,42 @@ def gen_scripts(
         nice_args = ""
     else:
         nice_args = " " + src_args
+
+    for env_file in info.env_files:
+        nice_args += f" --env-file '{env_file}'"
+
     restore = work_dir / "restore.sh"
     restore.write_text(
         template_local(
             "restore.sh",
             PROJECT_NAME=info.name,
-            SOURCE_ARGS=nice_args,
+            SOURCE_ARGS=nice_args
         )
     )
     make_executable(restore)
 
 
+def copy_env_file(source: Path, target: Path, work_dir: Path):
+    """
+    Copy an env file from one location to another.
+    """
+    target.parent.mkdir(exist_ok=True, parents=True)
+    print(
+        "found env file",
+        source.name,
+        "- copying to",
+        target.relative_to(work_dir),
+    )
+    fs_copy(source.name, target)
+
+
 def backup(
     project_name: str,
     output: Path,
-    password: Union[str, None] = None,
+    password: str,
```

Comment:
> Password can be None if not set

Currently dkp doesn't work without a password:

```
$ ../dkp/dkp.py -o test.pack test-data
Traceback (most recent call last):
  File "/home/hcs/r/dkp/test-data/../dkp/dkp.py", line 521, in <module>
    main()
  File "/home/hcs/r/dkp/test-data/../dkp/dkp.py", line 506, in main
    args.passphrase is not None and args.passphrase != ""
                                    ^^^^^^^^^^^^^^^^^^^^^
AssertionError: Passphrase is not set via argument neither via PASSPHRASE environment variable
```

If the intention is to work without a password (which would be nice actually), we would need to modify the `backup` function not to call `encrypt` when `password` is `None`:

```diff
-       print("encrypting...")
-       encrypted_file = output.with_name(archive_file.name + ".gpg")
-       encrypt(archive_file, encrypted_file, password)
-       # replace archive
-       fs_move(encrypted_file, archive_file)
+       if password is not None:
+           print("encrypting...")
+           encrypted_file = output.with_name(archive_file.name + ".gpg")
+           encrypt(archive_file, encrypted_file, password)
+           # replace archive
+           fs_move(encrypted_file, archive_file)
```

Shall I implement this? I wouldn't change my existing commit (because that is in-line with the fact that currently dkp makes password mandatory). I would to it in a separate commit.

Reply:
Thanks! Yes, it supposed to work without password 😅